repository/lib/rom/repository/class_interface.rb

```diff
@@ -113,7 +113,13 @@ def define_restricted_command_method(type, views, **opts)
             if changeset.respond_to?(:commit)
               changeset.commit
             else
-              root.command(type, **opts).public_send(view_name, *view_args).call(*input)
+              #   in case of view_args been coercible to an array we do not want to splat it
+              #   example: Date and Time objects
+              if view_args.respond_to?(:to_a)
+                root.command(type, **opts).public_send(view_name, view_args).call(*input)
```

Comment:
After some investigation and trial error, relaying in the `arity` is not going to be that simple.

Trying to access the arity value from `method(view_name).arity` since `view` is `AutoCurry` the arity is always `-1` .
Playing a little bit more I found a way to access the `arity` through out `Rom::Relation::Curried`

How I get the Curried Relation by executing this line:
`root.command(type, **opts).public_send(view_name)`

I get a Curried Relation, which happens to have an arity public API.

But I notice a strange behavior if the custom method from the relation have default values like this:

```ruby
def expired(expiration_time=Time.now)
  where{ created_at < expiration_time }
end
```
Instead of returning a  `Rom::Relation::Curried` I get a `ROM::Commands::Composite` which do not have a public api for `arity`

@solnic is there something I'm missing?

Also just for fun I played a bit with arity from ruby
```ruby
class C
 def one;    end
 def two(a); end
 def three(a=1); end
 def four(*a);  end
 def five(a, b); end
 def six(a, b, *c);    end
 def seven(a, b, *c, &d); end
end

c = C.new

c.method(:one).arity
=> 0
c.method(:two).arity
=> 1
c.method(:three).arity
=> -1
c.method(:four).arity
=> -1
c.method(:five).arity
=> 2
c.method(:six).arity
=> -3
c.method(:seven).arity
=> -3
```